spec.bs

```diff
@@ -252,19 +252,218 @@ dictionary AuctionAdConfig {
+1. If |url| is an error, then return failure.
+1. Return |url|'s [=url/origin=].
+
+To <dfn>start bidding and scoring</dfn> given an [=auction config=] |auctionConfig|:
+1. If |auctionConfig|'s [=auction config/component auctions=] are empty:
+  1. TODO: Create seller worklet.
+1. OtherWise:
+  1. [=list/For each=] |component| in |auctionConfig|'s [=auction config/component auctions=]:
+    1. [=Start bidding and scoring=] with |component|.
+  1. TODO: If the last |component|'s seller worklet is received, create seller worklet.
+1. Create bidder worklets [=in parallel=]. [=list/For each=] |buyer| in
+  [=auction config/interest group buyers=]:
+  1. Let |bidderWorklet| be a [=bidder worklet=] object for |buyer|. TODO: expand.
+  1. Let |biddingScript| be the result of [=fetching a bidding javascript=] with |bidderWorklet|.
+  1. If |biddingScript| is an error, [=iteration/continue=].
+  1. TODO: Run all [=generate a bid|GenerateBid()=] tasks that are ready.
+  1. TODO: Maybe rewrite this part.
+
+To <dfn>create a request</dfn> given a [=url=] |url|, and a [=string=] <dfn for="create a request">
+|accept|</dfn>:
+1. Let |request| be a new [=request=] with the following properties:
+    :   [=request/URL=]
+    ::  |url|
+    :   [=request/header list=]
+    ::  A new [=header list=] containing a [=header=] named "`Accept`" whose value is |accept|
+    :   [=request/client=]
+    ::  `null`
+    :   [=request/window=]
+    ::  "`no-window`" TODO: verify
+    :   [=request/service-workers mode=]
+    ::  "`none`"
+    :   [=request/origin=]
+    ::  [=opaque origin=]
+    :   [=request/referrer=]
+    :: "`no-referrer`"
+    :   [=request/mode=]
+    ::  "`cors`"
+    :   [=request/credentials mode=]
+    ::  "`omit`"
+    :   [=request/cache mode=]
+    ::  "`no-store`"
+    :   [=request/redirect mode=]
+    :: "`error`"
+1. Return |request|.
+
+To <dfn>fetch a bidding javascript</dfn> with given [=bidder worklet=] |bidderWorklet|:
+1. Let |request| be the result of [=creating a request=] with |bidderWorklet|'s
+    [=bidder worklet/bidding url=], and [=create a request/accept=] set to
+    "`application/javascript`".
+1. [=Fetch=] |request|.
+1. Let |response| be the result of [=fetch=] when it asynchronously completes.
+1. Let |body| be |response|’s [=response/body=].
+1. If |body| is null, return failure.
+1. Let |headers| be |response|'s [=response/header list=].
+1. If [=header list/getting a structured field value=] "X-Allow-FLEDGE" from |headers| does not
+  return true, return failure.
+1. Let |mimeType| be the result of [=header list/extracting a MIME type=] from |headers|.
+1. If |mimeType| is failure or not a [=JavaScript MIME type=], return failure.
+1. If |mimeType|'s [=MIME type/parameters=]["`charset`"] [=map/exists=], and is not "utf-8" or
+  "us-ascii", return failure.
+1. Let |biddingScript| be the result of compiling |body|. If |biddingScript| is an error, return

```

Comment:
For us, it's [v8 script compiler](https://crsrc.org/c/content/services/auction_worklet/auction_v8_helper.cc;drc=38bff96f36fb3980fef273c1066ba65a7b32da7c;l=507). Didn't find a link, but it seems we don't need this in the spec, and just return |body| as bidding script?

Reply:
I approved it like this because you already have a TODO to actually start the Worklet, and when you fill that in, you'll be reading the other specs that start Worklets and can copy the way they prepare the script too. There's no need to get it perfect in each PR.